packages/fiber/src/core/store.ts

```diff
@@ -217,20 +217,19 @@ const createStore = (
       set((state) => ({ performance: { ...state.performance, current } }))
 
     // Handle frame behavior in WebXR
-    const handleXRFrame = (timestamp: number) => {
+    const handleXRFrame: THREE.XRFrameRequestCallback = (timestamp: number, frame?: THREE.XRFrame) => {
       const state = get()
       if (state.frameloop === 'never') return
 
-      advance(timestamp, true)
+      advance(timestamp, true, state, frame)
     }
 
     // Toggle render switching on session
     const handleSessionChange = () => {
       gl.xr.enabled = gl.xr.isPresenting
-      gl.setAnimationLoop(gl.xr.isPresenting ? handleXRFrame : null)
 
-      // If exiting session, request frame
-      if (!gl.xr.isPresenting) invalidate(get())
+      // @ts-expect-error
+      gl.xr.setAnimationLoop(gl.xr.isPresenting ? handleXRFrame : null)
```

Comment:
Looks like WebXRManager's types aren't correct for this method. `WebGLRenderer#setAnimationLoop` and `WebXRManager#setAnimationLoop` should be identical and accept `THREE.XRFrameRequestCallback | null`.